lua/telescope/previewers.lua

```diff
@@ -443,42 +541,102 @@ previewers.vimgrep = defaulter(function(opts)
   return previewers.new_buffer_previewer {
     setup = function()
-      return {}
+      return {
+        last_set_bufnr = nil
+      }
     end,
 
+    teardown = function(self)
+      if self.state and self.state.hl_id then
+        vim.api.nvim_buf_clear_namespace(self.state.last_set_bufnr, previewer_ns, 0, -1)
+      end
+    end,
+
+    define_preview = function(self, entry, status)
+      with_preview_window(status, nil, function()
+        vim.api.nvim_buf_clear_namespace(self.state.last_set_bufnr, previewer_ns, 0, -1)
+
+        local lnum = entry.lnum or 0
+        local path = from_entry.path(entry, true)
+        if path == nil then return end
+
+        file_maker(path, self.state.bufnr, function()
+          -- TODO(conni2461): We are sometimes outside of the buffer. Especially if we scroll through the list fast. hmm
+          if entry.lnum then
+            vim.api.nvim_buf_add_highlight(self.state.bufnr, self.state.winid, "Visual", entry.lnum - 1, 0, -1)
+            vim.api.nvim_win_set_cursor(status.preview_win, {lnum, 0})
+          end
+
+          self.state.last_set_bufnr = bufnr
+        end)
+      end)
+    end
+  }
+end, {})
+
+previewers.ctags = defaulter(function(_)
+  return previewers.new_buffer_previewer {
     teardown = function(self)
       if self.state and self.state.hl_id then
         pcall(vim.fn.matchdelete, self.state.hl_id, self.state.hl_win)
         self.state.hl_id = nil
       end
     end,
 
-    preview_fn = function(self, entry, status)
+    define_preview = function(self, entry, status)
       with_preview_window(status, nil, function()
         local scode = string.gsub(entry.scode, '[$]$', '')
         scode = string.gsub(scode, [[\\]], [[\]])
         scode = string.gsub(scode, [[\/]], [[/]])
         scode = string.gsub(scode, '[*]', [[\*]])
 
-        local new_bufnr = vim.fn.bufnr(entry.filename, true)
-        vim.fn.bufload(new_bufnr)
-
-        vim.api.nvim_win_set_buf(status.preview_win, new_bufnr)
-        vim.api.nvim_win_set_option(status.preview_win, 'wrap', false)
-        vim.api.nvim_win_set_option(status.preview_win, 'winhl', 'Normal:Normal')
-        vim.api.nvim_win_set_option(status.preview_win, 'signcolumn', 'no')
-        vim.api.nvim_win_set_option(status.preview_win, 'foldlevel', 100)
-        vim.api.nvim_win_set_option(status.preview_win, 'scrolloff', 999)
+        file_maker(entry.filename, self.state.bufnr, function()
+          pcall(vim.fn.matchdelete, self.state.hl_id, self.state.winid)
+          vim.cmd "norm! gg"
+          -- TODO(conni2461): Same problem. Search does no longer work.
+          vim.fn.search(scode)
 
-        pcall(vim.fn.matchdelete, self.state.hl_id, self.state.hl_win)
-        vim.cmd "norm! gg"
-        vim.fn.search(scode)
-
-        self.state.hl_win = status.preview_win
-        self.state.hl_id = vim.fn.matchadd('Search', scode)
+          self.state.hl_id = vim.fn.matchadd('Search', scode)
+        end)
```

Comment:
Same as comment above. `vim.fn.search(scode)` can't find the code because i think the lines are not yet set. Is there some magical code with which i can wait until the lines are set?